lib/order.rb

```diff
@@ -0,0 +1,22 @@
+require "./lib/modules/percentage"
+
+class Order
+  attr_reader   :name, :cost
+  attr_accessor :price, :discount
+  
+  def initialize(options = {})
+    @name     = options[:name] 
+    @price    = options[:price]
+    @discount = options[:discount]
+  end
+
+  def cost 
+    @discount ? @price - (@discount.percent_of @price) : @price
```

Comment:
I would split the first part of this ternary into a private method that explains what you're doing here